From: Andreas Gruenbacher <agruen@suse.de>
Subject: Fix building external modules on ppc32

We are setting up sources for building external modules like this:

  /usr/src/linux-obj> # create a .config file
  /usr/src/linux-obj> make -C /usr/src/linux O=$PWD oldconfig
  /usr/src/linux-obj> make -C /usr/src/linux O=$PWD prepare
  /usr/src/linux-obj> make -C /usr/src/linux O=$PWD scripts
  /usr/src/linux-obj> make -C /usr/src/linux O=$PWD clean

After that, external modules can be built with:

  /usr/src/module> make -C /usr/src/linux-obj M=$PWD

This fails for ppc32 because the `make clean' removes the
arch/powerpc/include directory. This should be done in
archmrproper instead of in archclean.

Signed-off-by: Andreas Gruenbacher <agruen@suse.de>

Index: linux-2.6.15/arch/powerpc/Makefile
===================================================================
--- linux-2.6.15.orig/arch/powerpc/Makefile
+++ linux-2.6.15/arch/powerpc/Makefile
@@ -167,6 +167,8 @@ endef
 
 archclean:
 	$(Q)$(MAKE) $(clean)=$(boot)
+
+archmrproper:
 	$(Q)rm -rf arch/$(ARCH)/include
 
 archprepare: checkbin
